envoy/network/dns_factory.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.pb.h"
+#include "envoy/extensions/filters/udp/dns_filter/v3alpha/dns_filter.pb.h"
+#include "envoy/extensions/network/dns_resolver/apple/v3/appl_dns_resolver.pb.h"
+#include "envoy/extensions/network/dns_resolver/cares/v3/cares_dns_resolver.pb.h"
+#include "envoy/network/dns.h"
+
+#include "source/common/runtime/runtime_features.h"
+
+namespace Envoy {
+namespace Network {
+
+class DnsResolverFactory : public Config::TypedFactory {
+public:
+  /**
+   * @returns a callback to create a DnsResolver.
+   */
+
+  virtual DnsResolverSharedPtr createDnsResolverCb(
+      Event::Dispatcher& dispatcher, const Api::Api& api,
+      const envoy::config::core::v3::TypedExtensionConfig& dns_resolver_config) PURE;
+
+  std::string category() const override { return "envoy.network_dnsresolvers"; }
+};
+
+// Retrieve the DNS related configurations in the passed in @param config, and store the data into
+// @param typed_dns_resolver_config. The design behavior is:
+//
+// 1) If the config has typed_dns_resolver_config, copy it into typed_dns_resolver_config and use
+// it.
+//
+// 2) Otherwise, synthetic a CaresDnsResolverConfig object for Envoy to use cares DNS library.
```

Comment:
Can you inline these comments with the source? It will be easier to follow. Thanks.

Reply:
ACK